rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -232,131 +233,51 @@ def build(self, view_requirements: ViewRequirementsDict) -> SampleBatch:
-            # Set of (probably non-consecutive) indices.
-            # Example:
-            #  shift=[-3, 0]
-            #  buffer=[-3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
-            #  resulting data=[[-3, 0], [-2, 1], [-1, 2], [0, 3], [1, 4], ...]
-            elif isinstance(view_req.shift, np.ndarray):
-                data = [
-                    d[self.shift_before + obs_shift + view_req.shift]
-                    for d in np_data[data_col]
-                ]
-            # Single shift int value. Use the trajectory as-is, and if
-            # `shift` != 0: shifted by that value.
-            else:
-                shift = view_req.shift + obs_shift
-
-                # Batch repeat (only provide a value every n timesteps).
-                if view_req.batch_repeat_value > 1:
-                    count = int(
-                        math.ceil(
-                            (len(np_data[data_col][0]) - self.shift_before)
-                            / view_req.batch_repeat_value
-                        )
-                    )
-                    data = [
-                        np.asarray(
-                            [
-                                d[
-                                    self.shift_before
-                                    + (i * view_req.batch_repeat_value)
-                                    + shift
-                                ]
-                                for i in range(count)
-                            ]
-                        )
-                        for d in np_data[data_col]
-                    ]
-                # Shift is exactly 0: Use trajectory as is.
-                elif shift == 0:
-                    data = [d[self.shift_before :] for d in np_data[data_col]]
-                # Shift is positive: We still need to 0-pad at the end.
-                elif shift > 0:
-                    data = [
-                        _to_float_np_array(
-                            np.concatenate(
-                                [
-                                    d[self.shift_before + shift :],
-                                    [
-                                        np.zeros(
-                                            shape=view_req.space.shape,
-                                            dtype=view_req.space.dtype,
-                                        )
-                                        for _ in range(shift)
-                                    ],
-                                ]
+
+                    # handle the case where the inds are out of bounds from the end
+                    element_at_t = []
+                    for index in inds:
+                        if index < len(d):
+                            element_at_t.append(d[index])
+                        else:
+                            element_at_t.append(
+                                np.zeros(
+                                    shape=view_req.space.shape,
```

Comment:
does this work for Dict spaces, for example? I have a feeling this may not work if obs_preprocessing is disabled?

Reply:
hmm good catch. Fixed.